3.assess_generalizability/scripts/generalizabilty_eval.py

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluate generalizability of the model across holdout plate(s)
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pandas as pd
+import pathlib
+from joblib import load
+import numpy as np
+from sklearn.metrics import (
+    accuracy_score,
+    precision_recall_curve,
+)
+from typing import Tuple
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+
+# ## Define helper functions
+
+# In[2]:
+
+
+def get_X_y_data(
+    df: pd.DataFrame, label: str, shuffle: bool = False
+) -> Tuple[pd.DataFrame, np.array]:
+    """Get X (feature space) and labels (predicting class) from pandas Data frame, including feature names.
+
+    Args:
+        df (pd.DataFrame): Data frame containing morphology.
+        label (str): Name of the Metadata column being used as the predicting class.
+        shuffle (bool, optional): Shuffle the feature columns to get a shuffled dataset. Defaults to False.
+
+    Returns:
+        Tuple[pd.DataFrame, np.array]: Returns DataFrame for the feature space (X) with feature names,
+                                       and np.array for the predicting class (y).
+    """
+    # Get the feature columns (excluding 'Metadata' columns and the label column)
+    feature_columns = [
+        col for col in df.columns if not col.startswith("Metadata") and col != label
+    ]
+
+    # Extract feature space (X) as a DataFrame (keep feature names)
+    X = df[feature_columns]
+
+    # Extract class label (y) as a NumPy array
+    y = df.loc[:, [label]].values
+    y = np.ravel(y)  # Flatten y to a 1D array
+
+    # If shuffle is True, shuffle the feature columns independently for training
+    if shuffle:
+        for column in X.T:
+            np.random.shuffle(column)
+
+    return X, y
+
+
+# ## Set paths and variables
+
+# In[ ]:
```

Comment:
This isn't executed?

Reply:
UGH there is some issue I am having with my notebooks not showing it is executed in the metadata so when nbconvert happens it doesn't work :( thanks for catching!